dgl/src/pugl/pugl.h

```diff
@@ -122,6 +122,140 @@ typedef enum {
+    PUGL_KC_C = 6,
+    PUGL_KC_D = 7,
+    PUGL_KC_E = 8,
+    PUGL_KC_F = 9,
+    PUGL_KC_G = 10,
+    PUGL_KC_H = 11,
+    PUGL_KC_I = 12,
+    PUGL_KC_J = 13,
+    PUGL_KC_K = 14,
+    PUGL_KC_L = 15,
+    PUGL_KC_M = 16,
+    PUGL_KC_N = 17,
+    PUGL_KC_O = 18,
+    PUGL_KC_P = 19,
+    PUGL_KC_Q = 20,
+    PUGL_KC_R = 21,
+    PUGL_KC_S = 22,
+    PUGL_KC_T = 23,
+    PUGL_KC_U = 24,
+    PUGL_KC_V = 25,
+    PUGL_KC_W = 26,
+    PUGL_KC_X = 27,
+    PUGL_KC_Y = 28,
+    PUGL_KC_Z = 29,
+    PUGL_KC_1 = 30,
+    PUGL_KC_2 = 31,
+    PUGL_KC_3 = 32,
+    PUGL_KC_4 = 33,
+    PUGL_KC_5 = 34,
+    PUGL_KC_6 = 35,
+    PUGL_KC_7 = 36,
+    PUGL_KC_8 = 37,
+    PUGL_KC_9 = 38,
+    PUGL_KC_0 = 39,
+    PUGL_KC_Escape = 41,
+    PUGL_KC_Delete = 42,
+    PUGL_KC_Tab = 43,
+    PUGL_KC_Space = 44,
+    PUGL_KC_Minus = 45,
+    PUGL_KC_Equals = 46,
+    PUGL_KC_LeftBracket = 47,
+    PUGL_KC_RightBracket = 48,
+    PUGL_KC_Backslash = 49,
+    PUGL_KC_Semicolon = 51,
+    PUGL_KC_Quote = 52,
+    PUGL_KC_Grave = 53,
+    PUGL_KC_Comma = 54,
+    PUGL_KC_Period = 55,
+    PUGL_KC_Slash = 56,
+    PUGL_KC_CapsLock = 57,
+    PUGL_KC_F1 = 58,
+    PUGL_KC_F2 = 59,
+    PUGL_KC_F3 = 60,
+    PUGL_KC_F4 = 61,
+    PUGL_KC_F5 = 62,
+    PUGL_KC_F6 = 63,
+    PUGL_KC_F7 = 64,
+    PUGL_KC_F8 = 65,
+    PUGL_KC_F9 = 66,
+    PUGL_KC_F10 = 67,
+    PUGL_KC_F11 = 68,
+    PUGL_KC_F12 = 69,
+    PUGL_KC_PrintScreen = 70,
+    PUGL_KC_ScrollLock = 71,
+    PUGL_KC_Pause = 72,
+    PUGL_KC_Insert = 73,
+    PUGL_KC_Home = 74,
+    PUGL_KC_PageUp = 75,
+    PUGL_KC_DeleteForward = 76,
+    PUGL_KC_End = 77,
+    PUGL_KC_PageDown = 78,
+    PUGL_KC_Right = 79,
+    PUGL_KC_Left = 80,
+    PUGL_KC_Down = 81,
+    PUGL_KC_Up = 82,
+    PUGL_KC_KP_NumLock = 83,
+    PUGL_KC_KP_Divide = 84,
+    PUGL_KC_KP_Multiply = 85,
+    PUGL_KC_KP_Subtract = 86,
+    PUGL_KC_KP_Add = 87,
+    PUGL_KC_KP_Enter = 88,
+    PUGL_KC_KP_1 = 89,
+    PUGL_KC_KP_2 = 90,
+    PUGL_KC_KP_3 = 91,
+    PUGL_KC_KP_4 = 92,
+    PUGL_KC_KP_5 = 93,
+    PUGL_KC_KP_6 = 94,
+    PUGL_KC_KP_7 = 95,
+    PUGL_KC_KP_8 = 96,
+    PUGL_KC_KP_9 = 97,
+    PUGL_KC_KP_0 = 98,
+    PUGL_KC_Point = 99,
+    PUGL_KC_NonUSBackslash = 100,
+    PUGL_KC_KP_Equals = 103,
+    PUGL_KC_F13 = 104,
+    PUGL_KC_F14 = 105,
+    PUGL_KC_F15 = 106,
+    PUGL_KC_F16 = 107,
+    PUGL_KC_F17 = 108,
+    PUGL_KC_F18 = 109,
+    PUGL_KC_F19 = 110,
+    PUGL_KC_F20 = 111,
+    PUGL_KC_F21 = 112,
+    PUGL_KC_F22 = 113,
+    PUGL_KC_F23 = 114,
+    PUGL_KC_F24 = 115,
+    PUGL_KC_Help = 117,
+    PUGL_KC_Menu = 118,
+    PUGL_KC_Mute = 127,
+    PUGL_KC_SysReq = 154,
+    PUGL_KC_Return = 158,
+    PUGL_KC_KP_Clear = 216,
+    PUGL_KC_KP_Decimal = 220,
+    PUGL_KC_LeftControl = 224,
+    PUGL_KC_LeftShift = 225,
+    PUGL_KC_LeftAlt = 226,
+    PUGL_KC_LeftGUI = 227,
+    PUGL_KC_RightControl = 228,
+    PUGL_KC_RightShift = 229,
+    PUGL_KC_RightAlt = 230,
+    PUGL_KC_RightGUI = 231
+} PuglKeyCodes;
```

Comment:
We might want to expose this outside of pugl too. Otherwise, the users will have to include `pugl.h` to use these keycodes.